### ISSUE TITLE
Bug: (False syntex error)

### ISSUE THREAD

**Issue by krishnaTORQUE** [eslint/eslint#17058]
### Environment

Node version: 18.15.0
npm version: 9.6.3
Local ESLint version: 8.37.0
Global ESLint version: 8.37.0
Operating System: Debian 11

### What parser are you using?

@typescript-eslint/parser

### What did you do?

`config.ts`
```typescript
const config: Record<string, string> = { backend: "http://127.0.0.1:3100" };
export default config;
```

`.eslintrc`
```
{
    "root": true,
    "env": {
        "es2021": true,
        "browser": true,
        "node": true,
        "commonjs": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/strict"
    ],
    "ignorePatterns": ["node_modules", "dist", "*.html", "*.css"],
    "overrides": [
        {
            "files": [
                "*.js",
                "*.jsx",
                "*.cjs",
                "*.mjs",
                "*.ts",
                "*.tsx",
                "*.svelte"
            ],
            "processor": "svelte3/svelte3",
            "parser": "@typescript-eslint/parser"
        }
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2021,
        "sourceType": "module",
        "parser": "@typescript-eslint/parser",
        "project": ["tsconfig.json"],
        "extraFileExtensions": [".svelte"]
    },
    "plugins": ["@typescript-eslint", "svelte3"],
    "rules": {
        "comma-dangle": "off",
        "@typescript-eslint/comma-dangle": [
            "error",
            {
                "arrays": "never",
                "objects": "always-multiline",
                "imports": "never",
                "exports": "never",
                "functions": "never"
            }
        ],
        "indent": ["error", 4],
        "linebreak-style": ["error", "unix"],
        "semi": ["error", "always"],
        "quotes": ["error", "double"],
        "quote-props": ["off", "consistent"],
        "no-await-in-loop": "error",
        "no-unreachable-loop": "error",
        "no-new-func": "error",
        "no-console": "error",
        "@typescript-eslint/array-type": "error",
        "@typescript-eslint/ban-types": "error",
        "@typescript-eslint/consistent-type-imports": "error",
        "@typescript-eslint/consistent-type-definitions": ["error", "type"],
        "@typescript-eslint/explicit-member-accessibility": "error",
        "@typescript-eslint/method-signature-style": "error",
        "@typescript-eslint/naming-convention": [
            "error",
            { "selector": "variableLike", "format": ["snake_case"] },
            { "selector": "typeLike", "format": ["PascalCase"] },
            { "selector": "method", "format": ["snake_case"] }
        ],
        "no-array-constructor": "off",
        "@typescript-eslint/no-array-constructor": "error",
        "@typescript-eslint/no-base-to-string": "error",
        "@typescript-eslint/no-confusing-non-null-assertion": "error",
        "@typescript-eslint/no-confusing-void-expression": "error",
        "no-dupe-class-members": "off",
        "@typescript-eslint/no-dupe-class-members": "error",
        "@typescript-eslint/no-duplicate-enum-values": "error",
        "@typescript-eslint/no-duplicate-type-constituents": "error",
        "@typescript-eslint/no-empty-interface": "error",
        "@typescript-eslint/no-extra-non-null-assertion": "error",
        "no-extra-semi": "off",
        "@typescript-eslint/no-extra-semi": "error",
        "@typescript-eslint/no-floating-promises": "off",
        "no-implied-eval": "off",
        "@typescript-eslint/no-implied-eval": "error",
        "@typescript-eslint/no-inferrable-types": "off",
        "@typescript-eslint/no-invalid-void-type": "error",
        "@typescript-eslint/no-meaningless-void-operator": "error",
        "@typescript-eslint/no-misused-new": "error",
        "@typescript-eslint/no-misused-promises": "error",
        "@typescript-eslint/no-mixed-enums": "error",
        "@typescript-eslint/no-namespace": "error",
        "@typescript-eslint/no-non-null-asserted-nullish-coalescing": "error",
        "@typescript-eslint/no-non-null-asserted-optional-chain": "error",
        "@typescript-eslint/no-non-null-assertion": "error",
        "@typescript-eslint/no-this-alias": "error",
        "no-throw-literal": "off",
        "@typescript-eslint/no-throw-literal": "error",
        "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error",
        "@typescript-eslint/no-unnecessary-condition": "error",
        "@typescript-eslint/no-unnecessary-type-arguments": "error",
        "@typescript-eslint/no-unnecessary-type-constraint": "error",
        "@typescript-eslint/no-unsafe-declaration-merging": "error",
        "@typescript-eslint/no-unsafe-return": "error",
        "no-unused-vars": "off",
        "@typescript-eslint/no-unused-vars": "error",
        "no-use-before-define": "off",
        "@typescript-eslint/no-use-before-define": "error",
        "no-useless-constructor": "off",
        "@typescript-eslint/no-useless-constructor": "error",
        "@typescript-eslint/no-useless-empty-export": "error",
        "@typescript-eslint/non-nullable-type-assertion-style": "error",
        "@typescript-eslint/prefer-string-starts-ends-with": "error",
        "@typescript-eslint/promise-function-async": "error",
        "@typescript-eslint/switch-exhaustiveness-check": "error"
    },
    "settings": {
        "svelte3/typescript": true
    }
}

```

**Error**
```shell
~/server/frontend-svelte/src/config.ts
1:50  error  Expected }  ParseError
```

### What did you expect to happen?

The error should not be there.

### What actually happened?

Its a false positive error.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Rec0iL99** [comment]
Hi @krishnaTORQUE, `typescript-eslint` is not maintained by the ESLint team. The best course of action here would be to open an issue in the [`typescript-eslint` repo](https://github.com/typescript-eslint/typescript-eslint/issues/new/choose). The maintainers over there would be able to help you better. Thanks.

**Comment by krishnaTORQUE** [comment]
Sorry about that.
Thanks
